3Dmol/glmodel.js

```diff
@@ -1993,7 +1993,35 @@ $3Dmol.GLModel = (function() {
          */
         this.selectedAtoms = function(sel, from) {
             var ret = [];
-            sel = sel || {};
+
+            // make a deep copy of an object. This does not support arbitrary
+            // javascript objects, but support enough for eveything that is
+            // used in selections: number, string, boolean, functions; as well
+            // as arrays and nested objects with values of the aformentioned
+            // types.
+            const deepCopy = function(object) {
```

Comment:
when selectedAtoms is recursively called, won't that result in an unnecessary additional copy?  You could create the cache of selectors as part of this function.  If the cache is already created, you know you don't need to do the copy again.

Reply:
I would indeed result in additional copies of the selection, which I think should be cheap enough to do. 

The issue with creating the caches here is that we would have two places in the code that have to be aware of caches: here and when the actual evaluation takes place. If you think the (small) added code complexity is worth the memory savings, I can implement it 😃